src/subprocess_test.cc

```diff
@@ -261,3 +261,17 @@ TEST_F(SubprocessTest, ReadStdin) {
   ASSERT_EQ(1u, subprocs_.finished_.size());
 }
 #endif  // _WIN32
+
+#ifndef _WIN32
+TEST_F(SubprocessTest, Background) {
+  time_t before = time(NULL);
+  Subprocess* subproc = subprocs_.Add("echo -n 123; sleep 10s&");
```

Comment:
The test will only take 10s if it fails, i.e. without the rest of the patch. The immediate subprocess will exit after backgrounding the sleep task, and so with my patch the exit will be observed and the test will complete immediately.